docs/extensions/redirects.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: MIT
+
+import json
+from pathlib import Path
+from typing import Dict
+
+from _types import SphinxExtensionMeta
+from sphinx.application import Sphinx
+
+api_redirect_source = """
```

Comment:
While this works really quite well, it's not ideal to embed the js source in the sphinx extension. Could this use a template instead?
From what I can tell that's mostly a matter of calling `copy_asset_file` with some context object that can then be used in the template, similar to what [hoverxref does](https://github.com/readthedocs/sphinx-hoverxref/blob/36c8cf07dbe89fe54d834e29c6b455a2be42b6df/hoverxref/extension.py#L83-L91).

Reply:
https://github.com/DisnakeDev/disnake/pull/722/commits/38d1094fae41306a3fd97635acbb0c32f43ce732